src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const results = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        if (action.keysToRemove && action.keysToRemove.length > 0) {
+          for (const key of action.keysToRemove) {
+            if (currentState.hasOwnProperty(key)) {
+              delete currentState[key];
+            }
+          }
+        }
```

Comment:
The check for `action.keysToRemove` and its length is unnecessary. Since you're iterating over `action.keysToRemove`, if it's empty or undefined, the loop won't run. You can safely remove this conditional check.

Reply:
You have to fix this comment